src/many-ledger/src/module.rs

```diff
@@ -305,18 +306,37 @@ impl ledger::LedgerCommandsModuleBackend for LedgerModuleImpl {
     }
 }
 
-impl ledger::LedgerTransactionsModuleBackend for LedgerModuleImpl {
-    fn transactions(
+impl module::events::EventsModuleBackend for LedgerModuleImpl {
+    fn info(
         &self,
-        _args: ledger::TransactionsArgs,
-    ) -> Result<ledger::TransactionsReturns, ManyError> {
-        Ok(ledger::TransactionsReturns {
-            nb_transactions: self.storage.nb_transactions(),
+        _args: module::events::InfoArgs,
+    ) -> Result<module::events::InfoReturn, ManyError> {
+        Ok(module::events::InfoReturn {
+            total: self.storage.nb_transactions(),
+            event_types: vec![
```

Comment:
Use `strum` `EnumIter`. It will allow you to `EventKind::iter().collect()`

Reply:
I'd like to, but that will require a change in `many-rs`. I'll do it, but would like to unblock this.